src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -81,6 +90,7 @@
     @Override
     public void beginTree(DetailAST rootAST) {
         classes = new ArrayDeque<>();
+        superClasses = new HashSet<>();
```

Comment:
Please move new instances to field's declaration.
Change these 2 lines to `clear`.